python/cuml/test/test_pickle.py

```diff
@@ -106,6 +106,8 @@ def test_regressor_pickle(tmpdir, datatype, model, data_size):
 
     cu_after_pickle_model = pickle_save_load(tmpdir, model)
 
+    del model
```

Comment:
Hmm... I thought the plan here was to delete the model before loading the pickled one so we can insure there is no leakage between them? It seem like we could put that in pickle_save_load just once (after pickling) and reduce the amount of code needed while also testing the load a bit more cleanly.

Reply:
updated the pickle_save_load function